data/preprocessor.js

```diff
@@ -0,0 +1,69 @@
+/* eslint-disable no-use-before-define */
+
+const Preprocessor = (jsonRaw = {}) => {
+  try {
+    const canRenderTimestamp =
+      // safely get deeply nested JSON values
+      get(['metadata', 'firstPublished'], jsonRaw) &&
```

Comment:
This can be `const firstPublished = get(['metadata', 'firstPublished'], jsonRaw)`, use `firstPublished` in the `canRenderTimestamp` assertion and then use it on like 16